addons/sale_project/models/project.py

```diff
@@ -394,20 +408,83 @@ def _get_revenues_items_from_sol(self, domain=None, with_action=True):
                             'type': 'object',
                             'args': json.dumps(args),
                         }
-                        other_revenues['action'] = action_params
+                        if len(sale_order_items) == 1:
+                            action_params['res_id'] = sale_order_items.id
+                        materials['action'] = action_params
         sequence_per_invoice_type = self._get_profitability_sequence_per_invoice_type()
         return {
             'data': [{'id': invoice_type, 'sequence': sequence_per_invoice_type[invoice_type], **vals} for invoice_type, vals in revenues_dict.items()],
             'total': {'to_invoice': total_to_invoice, 'invoiced': total_invoiced},
         }
 
+    def _get_aal_domain(self):
+        """
+            Taking only the aal where the sale_line_ids is not enough, for some invoices are related to the project, but not the sales order.
+            Those invoices are already computed in the revenues section, and therefore should not be taken into account. The only way to sort the invoices out
+            is with the move lines.
+        """
+        move_line_ids_no_sol = list(self.env['account.move.line']._search([('sale_line_ids', '=', False)]))
+        return [('account_id', '=', self.analytic_account_id.id), ('move_line_id', 'in', move_line_ids_no_sol + [False])]
+
+    def _get_items_from_aal(self, with_action=True):
+        def get_action(record_ids):
+            args = [[('id', 'in', record_ids)]]
+            if len(record_ids) == 1:
+                args.append(record_ids[0])
+            return {'name': 'action_aal', 'type': 'object', 'args': json.dumps(args)}
+
+        domain = self._get_aal_domain()
+        aal_other_search = self.env['account.analytic.line'].search(domain)
+        if not aal_other_search:
+            return None
+        # The account.groupe_account_readonly is set in the account_accountant module. There's no bridge between this module and the sale_project.
+        # Therefore we add this security check-up here because the use case can not be made without the accounting module.
+        can_see_aal = with_action and len(self) == 1 and self.user_has_groups('account.group_account_readonly')
+        total_revenues = total_costs = 0.0
+        costs = revenues = self.env['account.analytic.line']
+        for aal in aal_other_search:
+            aal_amount = aal.amount
+            if aal_amount < 0.0:
+                total_costs += aal_amount
+                costs |= aal
+            else:
+                total_revenues += aal_amount
+                revenues |= aal
+        costs_action = None
+        revenues_action = None
+        if can_see_aal:
```

Comment:
```suggestion
        total_revenues = total_costs = 0.0
        costs = revenues = self.env['account.analytic.line']
        for aal in aal_other_search:
            aal_amount = aal.amount
            if aal_amount < 0.0:
                total_costs += aal_amount
                costs |= aal
            else:
                total_revenues += aal_amount
                revenues |= aal
        costs_action = None
        revenues_action = None
        if with_action and len(self) == 1 and self.user_has_groups('account.group_account_readonly'):
```
As `can_see_aal` is only used once :)